src/number.cr

```diff
@@ -240,6 +240,16 @@ struct Number
     self
   end
 
+  # Returns `true` if value is equal to zero.
+  #
+  # ```
+  # 0.zero? #=> true
+  # 5.zero? #=> false
+  # ```
+  def zero? : Bool
+    self == typeof(self).zero
```

Comment:
For `struct Number` is ok the `typeof(self).zero` since the literal expanded would be of exact type and that is better. But on Complex and Timespan I would say to go for the plain/`@ivar == ` approach. The compiler will end up inlining and simplifying it, but I think is simpler the `@ivar ==` and we avoid the extra work in the optimization.